src/mumble/ConnectDialog.cpp

```diff
@@ -592,13 +593,20 @@ void ServerItem::hideCheck() {
 			hide = (uiUsers == 0);
 	}
 	if (hide != ishidden) {
-		if (hide)
+		if (hide || !nameMatchesFilter())
```

Comment:
In which situations is `hideCheck` called? Aka: Why does it need to know about the filter (after all the filtered entries are explicitly set to be hidden).

Reply:
`hideCheck` is definitely not the right spot. I'll move the check to `ServerItem::addServerItem`.